src/azul/chalice.py

```diff
@@ -120,11 +120,17 @@ def spec(self) -> JSON:
         }
 
     def self_url(self, endpoint_path=None, **params) -> str:
-        url = furl(scheme=self.current_request.headers.get('x-forwarded-proto', 'http'),
-                   netloc=self.current_request.headers['host'],
-                   path=endpoint_path or self.current_request.context['path'],
-                   args={k: v for k, v in sorted(params.items())})
-        return url.url
+        if self.current_request is None:
+            url = furl(config.service_endpoint())
+            scheme, netloc = url.scheme, url.netloc
+        else:
+            scheme = self.current_request.headers.get('x-forwarded-proto', 'http')
```

Comment:
can we update the default `http` to `https`? 
might be a parking lot discussion to have w/ @hannes-ucsc